packages/webpack-extraction-plugin/src/sortCSSRules.ts

```diff
@@ -0,0 +1,82 @@
+import { getStyleBucketName, GriffelRenderer, StyleBucketName, styleBucketOrdering } from '@griffel/core';
+import { compile, Element, KEYFRAMES, MEDIA, RULESET, serialize, stringify, SUPPORTS, tokenize } from 'stylis';
+
+export function getSelectorFromElement(element: Element) {
+  return tokenize(element.value).slice(1).join('');
+}
+
+export function getElementMetadata(element: Element): string {
+  if (element.type === MEDIA) {
+    return element.value.replace(/^@media/, '').trim();
+  }
+
+  return '';
+}
+
+export function getElementReference(element: Element, suffix = ''): string {
+  if (element.type === RULESET || element.type === KEYFRAMES) {
+    return element.value + suffix;
+  }
+
+  if (Array.isArray(element.children) && element.children.length === 1) {
+    return getElementReference(element.children[0], element.value + suffix);
+  }
+
+  function removeRootProperty(element: Element): unknown {
+    return {
+      ...element,
+      children: Array.isArray(element.children)
+        ? element.children.map(child => removeRootProperty(child))
+        : element.children,
+      root: undefined,
+      parent: undefined,
+    };
+  }
+
+  throw new Error(
+    [
+      'getElementReference(): An unhandled case, please report if it happens and provide debug information about an element:',
+      JSON.stringify(removeRootProperty(element), null, 2),
+    ].join('\n'),
+  );
+}
+
+export function getStyleBucketNameFromElement(element: Element): StyleBucketName {
+  if (element.type === KEYFRAMES) {
+    return 'k';
+  }
+
+  return getStyleBucketName(
+    getSelectorFromElement(element),
+    element.type === '@layer' ? element.value : '',
+    element.type === MEDIA ? element.value : '',
+    element.type === SUPPORTS ? element.value : '',
+  );
+}
+
+export function sortCSSRules(css: string, compareMediaQueries: GriffelRenderer['compareMediaQueries']): string {
+  return serialize(
+    compile(css)
+      .map(element => ({
+        ...element,
+        bucketName: getStyleBucketNameFromElement(element),
+        metadata: getElementMetadata(element),
```

Comment:
I think this metadata should follow the same object pattern that is used in the DOM renderer.

```ts
{ m: '(max-height: 1px)' }
```

That way adding metadata for core rules in the future will be straightfoward to translate to the extraction logic.

Alternative some kind of metadata parser pattern would be good because you can share that with the extraction implementation and core. However with currently only one rule with one metadata property it's not currently necessary

Reply:
Well, we don't have it typed, so for now I will leave it as it. As you said there is a single usage currently...